### PR TITLE
Perfect division applied for fee distribution

### PR DESCRIPTION
**Pull request by Anurupa03**

### CODE REVIEW

File: core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java [hunk 1]
```diff
@@ -32,13 +32,15 @@ public AbstractFeeDistribution(Address addressProvider) {
...
         Address daoFundAddr = getAddress(Contracts.DAO_FUND.getKey());
+
+        BigInteger remaining = amount;
+        BigInteger denominator = ICX;
+
         for (Address receiver : feeDistributionWeight.keySet()) {
             BigInteger percentageToDistribute = feeDistributionWeight.get(receiver);
-            BigInteger amountToDistribute = (percentageToDistribute.multiply(amount)).divide(ICX);
-            remaining = remaining.subtract(amountToDistribute);
+            BigInteger amountToDistribute = (percentageToDistribute.multiply(remaining)).divide(denominator);
```
Comment (goldworm-icon) [inline core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java:43]: Zero denominator can raise "java.lang.ArithmeticException: BigInteger divide by zero".

File: core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java [hunk 2]
```diff
@@ -32,13 +32,15 @@ public AbstractFeeDistribution(Address addressProvider) {
 
 
     protected void distributeFee(BigInteger amount) {
```
Comment (goldworm-icon) [inline core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java:34]:
```suggestion
    protected void distributeFee(BigInteger amount) {
        if (amount.signum() <= 0) {
            return;
        }
```

File: core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java (hunk 1 above)
Comment (goldworm-icon) [reply core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java:43]: I left my suggestion on the following issue:
https://github.com/ommfinance/openmoneymarket-audit/issues/16#issuecomment-1864943798

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

**Review by goldworm-icon — APPROVED** [review]

**Review by Suyog007 — APPROVED** [review]
Looks Good.

**Pull request merged**